src/api/CertificateManagement.ts

```diff
@@ -0,0 +1,184 @@
+                if (error.response) {
+                    if (
+                        await handleAuthenticationError({
+                            statusCode: error.response.status,
+                            error: error.response.data as APIError,
+                            returnValue: {} as Certificate,
+                            networkError: false,
+                        })
+                    ) {
+                        return await this.getCertificate(username);
+                    }
+                    return {
+                        returnValue: {} as Certificate,
+                        statusCode: error.response.status,
+                        error: error.response.data as APIError,
+                        networkError: false,
+                    };
+                } else {
+                    return {
+                        returnValue: {} as Certificate,
+                        statusCode: 0,
+                        error: {} as APIError,
+                        networkError: true,
+                    };
+                }
+            });
+    }
+
+    async getEnrollmentId(username: string): Promise<APIResponse<EnrollmentId>> {
+        return await this._axios
+            .get(`/certificates/${username}/enrollmentId`)
+            .then((response: AxiosResponse) => {
+                return {
+                    returnValue: response.data as EnrollmentId,
+                    statusCode: response.status,
+                    error: {} as APIError,
+                    networkError: false,
+                };
+            })
+            .catch(async (error: AxiosError) => {
+                if (error.response) {
+                    if (
+                        await handleAuthenticationError({
+                            statusCode: error.response.status,
+                            error: error.response.data as APIError,
+                            returnValue: {} as EnrollmentId,
+                            networkError: false,
+                        })
+                    ) {
+                        return await this.getEnrollmentId(username);
+                    }
+                    return {
+                        returnValue: {} as EnrollmentId,
+                        statusCode: error.response.status,
+                        error: error.response.data as APIError,
+                        networkError: false,
+                    };
+                } else {
+                    return {
+                        returnValue: {} as EnrollmentId,
+                        statusCode: 0,
+                        error: {} as APIError,
+                        networkError: true,
+                    };
+                }
+            });
+    }
+
+    async getEncryptedPrivateKey(username: string): Promise<APIResponse<EncryptedPrivateKey>> {
+        return await this._axios
+            .get(`/certificates/${username}/privateKey`)
```

Comment:
I'm assuming the backend has a check that the requestor is actually permitted to retrieve the private key even though the key is encrypted. Am I right?

Reply:
You are right. Authentication is handled by HTTP cookies, so you don't find that in our javascript code. ;)

Nobody except the user may retrieve the key via this endpoint, not even admins.

Reply:
Perfect.. that's how it should be.